### DISCUSSION TITLE
Add support to configure the modifier separator

### DISCUSSION THREAD

**Discussion by jgornick** [dcastil/tailwind-merge#167] [Ideas] [answered]
We've configured our classes to use a different [modifier separator](https://tailwindcss.com/docs/configuration#separator). Would it be possible to configure the separator with tailwind-merge?

If I'm understanding the flow of the code, it seems like we would only need to provide the configuration option to: https://github.com/dcastil/tailwind-merge/blob/e9f1e1b172e093f9a0e80afb58a6f0826ad59f9a/src/lib/merge-classlist.ts#L77-L112

Any help would be greatly appreciated! Thank you!

**Comment by dcastil** [comment]
Hey @jgornick! 👋 I think that should be possible. Are you using a single character as separator or multiple characters?

Just bookmarking some links from my research in here:
- Tailwind splits out modifieders in https://github.com/tailwindlabs/tailwindcss/blob/226be119dfb95706486df0d38f3ce357fb4ac856/src/lib/generateRules.js#L534.
- Tailwind's split modifier logic is in https://github.com/tailwindlabs/tailwindcss/blob/226be119dfb95706486df0d38f3ce357fb4ac856/src/util/splitAtTopLevelOnly.js.
- There might be a bug in my current split modifier logic. Need to read more about it in https://github.com/tailwindlabs/tailwindcss/pull/9320.

**Reply by dcastil** [reply, marked as answer]
@jgornick I just added support for custom separators in [v1.8.0](https://github.com/dcastil/tailwind-merge/releases/tag/v1.8.0).

You can add it to the tailwind-merge config like this:

```ts
const twMerge = extendTailwindMerge({ separator: '…' })
```

Let me know if you encounter any issues with it!

**Reply by jgornick** [reply]
Well that was fast :) 

To answer your earlier question, we use 2 characters (`__`). It looks like the implementation does support multiple characters so we should be all good!

Thank you so much for this update!

**Reply by dcastil** [reply]
Yeah, I basically copied the modifier separation logic from Tailwind, so it should work the same way. 😊
